scripts/populate_expressions_list.py

```diff
@@ -0,0 +1,216 @@
+    elif f_name.startswith("$"):
+        syntax += f"{f_name}\n"
+        arguments = ''
+    else:
+        syntax += f"{f_name}()\n"
+        arguments = ''
+
+    if optional_args:
+        syntax += '\n       [] marks optional arguments\n'
+
+    if 'type' in function_dict and function_dict['type'] == 'expression':
+        syntax = ''
+
+    # Prepare examples
+    if 'examples' in function_dict:
+        ex_list = []
+        for ex in function_dict['examples']:
+            example = f"{ex['expression']} → {ex['returns']}"
+            if 'note' in ex:
+                example += f"\n\n         {sphynxify_html(ex['note'])}"
+
+            ex_list.append(example)
+
+        examples = "   * - Examples\n     - * " + "\n       * ".join(ex_list)
+    else:
+        examples = ''
+
+    if 'variant' in function_dict:
+        v_description = (f"**{function_dict['variant']}**\n\n"
+                         f"{function_dict['variant_description']}\n\n")
+    else:
+        v_description = ''
+
+    if 'notes' in function_dict:
+        notes = f"\n\n.. note:: {function_dict['notes']}"
+    else:
+        notes = ''
+
+    text = (f"{v_description}"
+            f".. list-table::\n"
+            f"   :widths: 15 85\n"
+            f"   :stub-columns: 1\n"
+            f"\n"
+            f"{syntax}"
+            f"{arguments}"
+            f"{examples}"
+            f"{notes}"
+            f"\n\n\n")
+    return text
+
+# Get a list of all files on that folder
+
+(_, _, filenames) = next(walk(help_folder))
+
+# Store groups and functions in memory
+groups = {}
+functions = {}
+for file in filenames:
+    filepath = path.join(help_folder, file)
+    with open(filepath) as f:
+        data = json.load(f)
+    #print(data)
+    if data['type'] == 'group':
+        groups[data['name']] = {'description': data['description'], 'func_list': []}
+    elif data['type'] == 'function' or data['type'] == 'expression':
+        functions[data['name']] = data
+    else:
+        print(f'{data["name"]} is not a group or a function. It is a {data["type"]}')
+
+# Distribute functions by the groups
+for f_name in functions.keys():
+    function = functions[f_name]
+    if 'groups' in function:
+        for g_name in function['groups']:
+            if g_name not in ('deprecated'):
+                groups[g_name]['func_list'].append(function['name'])
+            else:
+                print(f'Warning: Function {function["name"]} is deprecated!')
+    else:
+        print(f'Warning: Function {function["name"]} has no group!')
+    #    print(group)
+
+for g_name in groups:
+    func_list = groups[g_name]['func_list']
+    if len(func_list) < 1:
+        continue
+    func_list.sort(key=lambda x: x.strip('$'))
+    output_group_file = path.join(output_folder, g_name.replace(' ','_') + '.rst')
+    with open(output_group_file, 'w') as f:
+        f.write(f".. DO NOT EDIT THIS FILE DIRECTLY. It is generated automatically by\n"
```

Comment:
```suggestion
        f.write(f":orphan:\n\n"
                f".. DO NOT EDIT THIS FILE DIRECTLY. It is generated automatically by\n"
```
This avoids `WARNING: document isn't included in any toctree`

Reply:
That is not needed. The folder is in the exclude paths in the conf.py

Reply:
I noticed that after I suggested this, but there should not be any harm in using `:orphan:`?